accesscontroltool-bundle/src/main/java/biz/netcentric/cq/tools/actool/helper/QueryHelper.java

```diff
@@ -73,32 +77,47 @@
                     }
                 }
             } catch (RepositoryException e) {
-                AcHelper.LOG.error("Exception: {}", e);
+                LOG.error("Exception: {}", e);
             }
         }
         Set<Node> nodes = new LinkedHashSet<Node>();
         try {
             // get the rep:policy node of "/", if existing
-            if (session.nodeExists("/rep:policy")) {
-                nodes.add(session.getNode("/rep:policy"));
+            if (session.nodeExists(ROOT_REP_POLICY_NODE)) {
+                nodes.add(session.getNode(ROOT_REP_POLICY_NODE));
             }
-            if (session.nodeExists("/" + Constants.REPO_POLICY_NODE)) {
-                nodes.add(session.getNode("/" + Constants.REPO_POLICY_NODE));
+            if (session.nodeExists(ROOT_REPO_POLICY_NODE)) {
+                nodes.add(session.getNode(ROOT_REPO_POLICY_NODE));
             }
 
             // get the rep:policy node of "/home", if existing
-            if (session.nodeExists("/home/rep:policy")) {
-                nodes.add(session.getNode("/home/rep:policy"));
+            if (session.nodeExists(HOME_REP_POLICY)) {
+                nodes.add(session.getNode(HOME_REP_POLICY));
             }
 
+            boolean indexForRepACLExists = session.nodeExists(OAK_INDEX_PATH_REP_ACL);
+            LOG.debug("Index for repACL exists: {}",indexForRepACLExists);
+            String queryForAClNodes = indexForRepACLExists ? 
+                    "SELECT * FROM [rep:ACL] WHERE ISDESCENDANTNODE([%s])" : 
+                    "SELECT ace.* FROM [rep:ACE] AS ace WHERE ace.[rep:principalName] IS NOT NULL AND ISDESCENDANTNODE(ace, [%s])";
+            LOG.debug("Query to obtain all ACLs: {}", queryForAClNodes);
```

Comment:
For now we leave the old query for the case  the index is there (to reduce the impact for existing users if there are problems)